File: dataset/human/python/file_8008.py
N = 4010
p = 998244353
mi2 = [0] * N
inv = [0] * N
fac = [0] * N

def C(x, y):
    return fac[x] * inv[y] * inv[x - y] % p

def Calc(n, a, b):
    ans = 0
    for i in range(min(a, n) + 1):
        if i == b == 0:
            continue
        ans += C(a, i) * mi2[i] * C(n + b - 1, n - i) % p
    return ans % p

mi2[0] = fac[0] = inv[0] = inv[1] = 1
for i in range(1, N):
    mi2[i] = mi2[i - 1] * 2 % p
    fac[i] = fac[i - 1] * i % p
for i in range(2, N):
    inv[i] = inv[p % i] * (p - p // i) % p
for i in range(2, N):
    inv[i] = inv[i - 1] * inv[i] % p

k, n = map(int, input().split())
for i in range(2, 2 * k + 1):
    ans = 0
    cnt = min(k, (i - 1) >> 1) - max(i - k, 1) + 1
    if i & 1:
        ans += Calc(n, cnt, k - cnt * 2)
    else:
        ans += Calc(n, cnt, k - cnt * 2 - 1)
        ans += Calc(n - 1, cnt, k - cnt * 2 - 1)
    print(ans % p)

File: dataset/human/c/file_8008.c
#include<stdio.h>
#define MAX 1000002
#define MOD 1000000007

long long fac[MAX], finv[MAX], inv[MAX];

void init() {
    fac[0] = fac[1] = 1;
    finv[0] = finv[1] = 1;
    inv[1] = 1;
    for (int i = 2; i < MAX; i++){
        fac[i] = fac[i - 1] * i % MOD;
        inv[i] = MOD - inv[MOD%i] * (MOD / i) % MOD;
        finv[i] = finv[i - 1] * inv[i] % MOD;
    }
}

long long f(int n1, int n2){
    return fac[n1+n2] * (finv[n1] * finv[n2] % MOD) % MOD;
}

int main()
{
  int x, y, a=0, b=0;
  scanf("%d %d", &x, &y);
  init();

  for(int i=0; i<=MAX; i++){
    int j = x - 2*i;
    if(y == i+2*j){
      a = i;
      b = j;
    }
  }

  if((x+y)%3 != 0){
    printf("0");
    return 0;
  }

  if(a<0 || b<0){
    printf("0");
    return 0;
  }

  if(a==0 && b==0){
    printf("0");
    return 0;
  }

  long long ans;
  ans = f(a, b);
  printf("%lld", ans);
  return 0;
}
